enterprise/backend/src/metabase_enterprise/data_editing/api.clj

```diff
@@ -438,247 +441,40 @@
-            row-delay   (delay
-                         ;; TODO this is incorrect - in general the row will not come from the table we are acting
-                         ;;      upon - this is not even true for our first use case!
-                          (when table-id
-                            (let [pk-fields    (data-editing/select-table-pk-fields table-id)
-                                  pk           (select-keys input (mapv (comp keyword :name) pk-fields))
-                                  pk-satisfied (= (count pk) (count pk-fields))]
-                              (when pk-satisfied
-                                (first (data-editing/query-db-rows table-id pk-fields [pk]))))))]
-        (cond
-          saved-id
-          (describe-saved-action :action-id              saved-id
-                                 :row-action-dashcard-id dashcard-id
-                                 :param-mapping          mapping
-                                 :row-delay              row-delay)
-
-          action-kw
-          (describe-table-action :action-kw     action-kw
-                                 :table-id      table-id
-                                 :param-mapping mapping
-                                 :dashcard-viz  (:dashcard-viz unified)
-                                 :row-delay     row-delay)
-
-          :else (ex-info "Not a supported row action" {:status-code 500, :scope scope, :unified unified})))
-      :else
-      (throw (ex-info "Not able to execute given action yet" {:status-code 500, :scope scope, :unified unified})))))
+  (let [scope   (actions/hydrate-scope scope)
+        ;; TODO consolidate this with [[fetch-unified-action]]
+        unified (cond
+                  (not (#{"table.row/create"
+                          "table.row/update"
+                          "table.row/delete"} action_id))
+                  (fetch-unified-action scope action_id)
+
+                  (:dashcard-id scope)
+                  (let [{:keys [dashcard-id]} scope
+                        {:keys [dashboard_id visualization_settings]} (t2/select-one :model/DashboardCard dashcard-id)]
+                    (api/read-check (t2/select-one :model/Dashboard dashboard_id))
+                    {:dashcard-viz  visualization_settings
+                     :inner-action  {:action-kw (keyword action_id)
+                                     :mapping   {:table-id (:table_id visualization_settings)
+                                                 :row      ::root}}
+                     ;; TODO: this should belongs to our configuration
+                     :param-mapping (->> visualization_settings
+                                         :editableTable.enabledActions
+                                         (some (fn [{:keys [id parameterMappings]}]
+                                                 (when (= id action_id)
+                                                   parameterMappings))))})
```

Comment:
Our migrate on read for this will be something like this

```suggestion
                                         (some (fn [{:keys [id configuration parameterMappings]}]
                                                 (when (= id action_id)
                                                   (or configuration parameterMappings)))))})
```